tests/test_issue_1514/test_usic.py

```diff
@@ -0,0 +1,303 @@
+import os
+import shutil
+import json
+import radical.utils as ru
+import radical.pilot as rp
+import saga as rs
+from radical.pilot.umgr.staging_input.default import Default
+import saga.filesystem as rsf
+import pytest
+import re
+import saga.utils.pty_shell as rsups
+
+
+try:
+    import mock
+except ImportError:
+    from unittest import mock
+
+# User Input for test
+#-----------------------------------------------------------------------------------------------------------------------
+resource_name = 'local.localhost'
+access_schema = 'ssh'
+#-----------------------------------------------------------------------------------------------------------------------
+
+
+# Extract info from RP config file
+#-----------------------------------------------------------------------------------------------------------------------
+config_loc = '../../src/radical/pilot/configs/resource_%s.json'%resource_name.split('.')[0]
+path_to_rp_config_file = os.path.realpath(os.path.join(os.getcwd(),config_loc))
+cfg_file = ru.read_json(path_to_rp_config_file)[resource_name.split('.')[1]]
+
+## Resolve environment variables in cfg
+if '$' in cfg_file['default_remote_workdir']:
+    target = cfg_file[access_schema]["filesystem_endpoint"].split('//')[1]
+
+    shell   = rsups.PTYShell('%s://%s'%(access_schema, target))
```

Comment:
you can just parse the file system endpoint here - no need to parse the URL and put is back together.  Also, URL parsing is tricky.

```
shell   = rsups.PTYShell(cfg_file[access_schema]["filesystem_endpoint"])
```

Or am I missing something?

Reply:
You're right. Why did I even do that :-|